### ISSUE TITLE
hub clone <owner>/<repo> fails when directory <owner> exists

### ISSUE THREAD

**Issue by ackalker** [github/hub#578]
I'm trying to clone the atom/apm repository while in a directory which already contains a directory named atom (and yes, this contains a clone of the atom/atom repo :-) ):
```
$ hub --version
git version 2.0.0
hub version 1.12.0
$ ls -d atom
atom
$ hub clone atom/apm
fatal: repository 'atom/apm' does not exist
$ hub clone atom/apm --
fatal: repository 'atom/apm' does not exist
$ hub clone -- atom/apm
fatal: repository 'atom/apm' does not exist
$ hub clone atom/apm apm
fatal: repository 'atom/apm' does not exist
$ hub clone -- atom/apm apm
fatal: repository 'atom/apm' does not exist
$ hub clone atom/apm -- apm
fatal: repository 'atom/apm' does not exist
$ hub clone atom/apm apm --
fatal: repository 'atom/apm' does not exist
```

From reading hub's documentation, there appears to be no way to distinguish the owner of a repository from an already existing directory.
Please implement a way of doing this without the need to create a temporary directory, clone into that, then move the clone into place:
```
$ mkdir tmp
$ cd tmp
$ hub clone atom/apm
Cloning into 'apm'...
remote: Reusing existing pack: 3837, done.
remote: Counting objects: 192, done.
remote: Compressing objects: 100% (190/190), done.
remote: Total 4029 (delta 116), reused 0 (delta 0)
Receiving objects: 100% (4029/4029), 4.48 MiB | 1.44 MiB/s, done.
Resolving deltas: 100% (2632/2632), done.
Checking connectivity... done.
$ mv apm ..
$ cd ..
$ rmdir tmp
$ ls -d atom apm
apm  atom
```

**Comment by ysaimanojkumar** [comment]
@ackalker :+1:
Also a new option can be added to indicate to hub that the "atom/apm" is actually a local directory.

**Comment by jingweno** [comment]
Can we first check that if `atom/apm` is a valid git directory? If it is, we then clone from a local directory. Otherwise it's cloning from a GitHub repo. Of course, printing out info about cloning from a GitHub repo will help make less confusion.

**Comment by mislav** [comment]
This is now fixed as of #1083!
